packages/pigeon/lib/dart_generator.dart

```diff
@@ -500,48 +512,48 @@ void generateDart(DartOptions opt, Root root, StringSink sink) {
       indent.write('Object encode() ');
       indent.scoped('{', '}', () {
         indent.writeln(
-          'final Map<Object?, Object?> pigeonMap = <Object?, Object?>{};',
+          'final List<Object?> pigeonList = <Object?>[];',
         );
         for (final NamedType field in klass.fields) {
-          indent.write("pigeonMap['${field.name}'] = ");
+          indent.write('pigeonList.add(');
           final String conditional = field.type.isNullable ? '?' : '';
           if (customClassNames.contains(field.type.baseName)) {
             indent.addln(
-              '${field.name}$conditional.encode();',
+              '${field.name}$conditional.encode());',
             );
           } else if (customEnumNames.contains(field.type.baseName)) {
             indent.addln(
-              '${field.name}$conditional.index;',
+              '${field.name}$conditional.index);',
             );
           } else {
-            indent.addln('${field.name};');
+            indent.addln('${field.name});');
           }
         }
-        indent.writeln('return pigeonMap;');
+        indent.writeln('return pigeonList;');
       });
```

Comment:
Looking at the generated output: we could avoid the need to build the array dynamically. Currently we're writing, e.g.:

```
    final List<Object?> pigeonList = <Object?>[];
    pigeonList.add(query);
    pigeonList.add(anInt);
    pigeonList.add(aBool);
    return pigeonList;
```

We could instead do:
```
  return <Object?>[
    query,
    anInt,
    aBool,
  ];
```

The generator code for that should actually be easier than what's here now (using a scoped for the array literal).

Reply:
It's been so long now, I can't remember why I chose to do it this way. I'll switch it for sure.